daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -546,7 +554,13 @@ trait ElementBaseGrammarMixin
   private def explicitBinaryLengthInBits() = {
     val lengthFromProp: JLong = repElement.lengthEv.optConstant.get
     val nbits = repElement.lengthUnits match {
-      case LengthUnits.Bits => lengthFromProp.longValue()
+      case LengthUnits.Bits => repElement.primType match {
+         case PrimType.Integer | PrimType.NonNegativeInteger => {
+           SDW(WarnID.DeprecatedTypeUse, "In a future release, lengthUnits='bits' will only be supported for integer types of 64 bits or less")
```

Comment:
Say lengthUnits 'bits' will not be supported for types xs:integer and xs:nonNegativeInteger.

Reply:
Updated message